salt/states/kmod.py

```diff
@@ -31,50 +60,99 @@ def present(name, persist=False):
 
     persist
         Also add module to ``/etc/modules``
+
+    mods
+        A list of modules to verify are loaded.  If this argument is used, the
+        ``name`` argument, although still required, is not used, and becomes a
+        placeholder
+
+        .. versionadded:: Boron
     '''
+    if not isinstance(mods, (list, tuple)):
+        mods = [name]
     ret = {'name': name,
            'result': True,
            'changes': {},
            'comment': ''}
-    mods = __salt__['kmod.mod_list']()
 
+    loaded_mods = __salt__['kmod.mod_list']()
     if persist:
         persist_mods = __salt__['kmod.mod_list'](True)
-        # Intersection of persist and non persist module
-        mods_set = list(set(mods) & set(persist_mods))
-    else:
-        mods_set = mods
-    if name in mods_set:
-        ret['comment'] = ('Kernel module {0} is already present'
-                          .format(name))
-        return ret
-    # Module is not loaded, verify availability
+        # Intersection of loaded modules and persistent modules
+        loaded_mods = list(set(loaded_mods) & set(persist_mods))
+
+    # Intersection of loaded and proposed modules
+    already_loaded = list(set(loaded_mods) & set(mods))
+    if len(already_loaded) == 1:
+        comment = 'Kernel module {0} is already present'.format(already_loaded[0])
+        _append_comment(ret, comment)
+    elif len(already_loaded) > 1:
+        comment = 'Kernel modules {0} are already present'.format(', '.join(already_loaded))
+        _append_comment(ret, comment)
+
+    if len(already_loaded) == len(mods):
+        return ret  # all modules already loaded
+
+    # Complement of proposed modules and already loaded modules
+    not_loaded = list(set(mods) - set(already_loaded))
+
     if __opts__['test']:
         ret['result'] = None
-        ret['comment'] = 'Module {0} is set to be loaded'.format(name)
+        if len(ret['comment']):
+            ret['comment'] += '\n'
+        if len(not_loaded) == 1:
+            comment = 'Kernel module {0} is set to be loaded'.format(not_loaded[0])
+        else:
+            comment = 'Kernel modules {0} are set to be loaded'.format(', '.join(not_loaded))
+        _append_comment(ret, comment)
         return ret
-    if name not in __salt__['kmod.available']():
-        ret['comment'] = 'Kernel module {0} is unavailable'.format(name)
+
+    # Complement of proposed, unloaded modules and available modules
+    unavailable = list(set(not_loaded) - set(__salt__['kmod.available']()))
+    if unavailable:
+        if len(unavailable) == 1:
+            comment = 'Kernel module {0} is unavailable'.format(unavailable[0])
+        else:
+            comment = 'Kernel modules {0} are unavailable'.format(', '.join(unavailable))
+        _append_comment(ret, comment)
         ret['result'] = False
-        return ret
```

Comment:
Cool. It just seemed like it might have been in error so I wanted to bring it to your attention. Sounds like you've carefully considered it and come to a sound decision. Thanks!